src/controllers/validators/IllnessEndDateValidator.ts

```diff
@@ -0,0 +1,60 @@
+import { Session } from 'ch-node-session-handler';
+import { Request } from 'express';
+import moment from 'moment';
+
+import { DateValidator } from 'app/controllers/validators/DateValidator';
+import { loggerInstance, loggingMessage } from 'app/middleware/Logger';
+import { Appeal } from 'app/models/Appeal';
+import { ApplicationData, APPLICATION_DATA_KEY } from 'app/models/ApplicationData';
+import { Illness } from 'app/models/Illness';
+import { SESSION_NOT_FOUND_ERROR } from 'app/utils/CommonErrors';
+import { ValidationError } from 'app/utils/validation/ValidationError';
+import { ValidationResult } from 'app/utils/validation/ValidationResult';
+
+export class IllnessEndDateValidator extends DateValidator {
+
+    private ILLNESS_END_DATE_BEFORE_ILLNESS_START_DATE: string = 'End date must be after the start date';
+
+    constructor() {
+        super();
+    }
+
+    public async validate(request: Request): Promise<ValidationResult> {
+
+        const dayField: string = 'day';
+        const monthField: string = 'month';
+        const yearField: string = 'year';
+        const dateField: string = 'date';
+
+        request.body.date = moment(`${request.body[yearField]}-${request.body[monthField]}-${request.body[dayField]}`)
+            .toDate();
+
+        const validationResult: ValidationResult = await super.validate(request);
+
+        const session: Session | undefined = request.session;
+
+        if (!session) {
+            throw SESSION_NOT_FOUND_ERROR;
+        }
+
+        const appData: ApplicationData = session.getExtraData<ApplicationData>(APPLICATION_DATA_KEY)
+            || { appeal: {} as Appeal, navigation: { permissions: [] } } as ApplicationData;
```

Comment:
if appData is undefined I will throw an Error instead, something like

    if(!appData) {
        throw APPLICATION_DATA_UNDEFINED;
    }

Reply:
Will do

Reply:
Just want to check. Line 40, doesn't this line always set appData? I'm guessing if the getExtraData returns nothing then an empty appData object is created?

Reply:
At this page `appData` should not be empty, better fail quick